hail/src/main/scala/is/hail/io/plink/LoadPlink.scala

```diff
@@ -186,14 +213,28 @@ object MatrixPLINKReader {
       colKey = Array("s"),
       colType = saSignature.virtualType,
       rowType = TStruct(
-        "locus" -> TLocus.schemaFromRG(referenceGenome),
+        "locus" -> locusType,
         "alleles" -> TArray(TString),
         "rsid" -> TString,
         "cm_position" -> TFloat64),
       rowKey = Array("locus", "alleles"),
       entryType = TStruct("GT" -> TCall))
+    assert(locusAllelesType == fullMatrixType.rowKeyStruct)
+
+    val partitioner = new RVDPartitioner(
+      locusAllelesType,
+      Array.tabulate[Interval](nPartitions) { i =>
+        val first = partScan(i)
+        val last = partScan(i + 1) - 1
+        assert(first < last)
+        Interval(
+          variants(first).locusAlleles,
+          variants(last).locusAlleles,
+          includesStart = true, includesEnd = true)
```

Comment:
I'm surprised this is inclusive on both ends. Is that right?

Reply:
No, you're quite right, the keys might be duplicate.  I'll fix it.

Reply:
I now scan the end of the partition forward and only break partitions along locus boundaries.  I still keep it inclusive on both ends.